### PR TITLE
Update tracy to 0.11 and improve developer experience

### PR DESCRIPTION
**Pull request by RyeMutt**
* Update to Tracy 0.11 and change integration method to in-source build via including `TracyClient.cpp`
* Copy windows profiler client to viewer package directory for better developer experience when tracy is enabled.
* Introduce build-time option `USE_TRACY_ON_DEMAND` to control profiler startup behavior
* Introduce build-time option `USE_TRACY_LOCAL_ONLY` to control profiler network broadcast behavior
* Introduce mutex instrumentation and do an initial pass on LLSingleton, LLInstanceTracker, and logging infrastructure

### CODE REVIEW

**Comment by Ansariel** [comment]
Nice!

**Review by marchcat — APPROVED** [review]
Thank you!

File: indra/cmake/Tracy.cmake [hunk 1]
```diff
@@ -4,14 +4,26 @@ include(Prebuilt)
...
+  option(USE_TRACY_ON_DEMAND "Use Tracy profiler." ON)
+  option(USE_TRACY_LOCAL_ONLY "Use Tracy profiler." OFF)
+
   use_system_binary(tracy)
   use_prebuilt_binary(tracy)
 
   target_include_directories( ll::tracy SYSTEM INTERFACE ${LIBS_PREBUILT_DIR}/include/tracy)
-  target_link_libraries( ll::tracy INTERFACE TracyClient )
+  
+  target_compile_definitions(ll::tracy INTERFACE -DTRACY_ENABLE=1 -DTRACY_ONLY_IPV4=1)
```
Comment (marchcat) [inline indra/cmake/Tracy.cmake:18]:
```suggestion

  target_compile_definitions(ll::tracy INTERFACE -DTRACY_ENABLE=1 -DTRACY_ONLY_IPV4=1)
```
To stop pre-commit from complaining.

**Comment by marchcat** [comment]
Also related to #1098.

**Pull request merged**